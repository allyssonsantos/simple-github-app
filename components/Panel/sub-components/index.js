import Header from './Header';
import Title from './Title';
import Stars from './Stars';

export { Header, Title, Stars };
