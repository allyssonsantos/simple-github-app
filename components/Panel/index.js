import React, { Component } from 'react';
import { Title, Stars, Header } from './sub-components';

class Panel extends Component {
  constructor(props) {
    super(props);
  }

  static Header = Header;
  static Title = Title;
  static Stars = Stars;

  render() {
    const { children } = this.props;
    return <>{children}</>;
  }
}

export default Panel;
