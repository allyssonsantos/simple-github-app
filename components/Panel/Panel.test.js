import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Panel from './';

describe('With Snapshot Testing', () => {
  it('Should match snapshot without compound components children', () => {
    const component = renderer.create(<Panel>This is a Panel</Panel>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should match snapshot Header, Stars and Title children components', () => {
    const PanelComponent = () => (
      <Panel>
        <Panel.Header>
          <Panel.Title>Title</Panel.Title>
          <Panel.Stars>Stars</Panel.Stars>
        </Panel.Header>
      </Panel>
    );
    const component = renderer.create(<PanelComponent />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
