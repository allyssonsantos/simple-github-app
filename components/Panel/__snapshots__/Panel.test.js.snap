// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Should match snapshot Header, Stars and Title children components 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f2f2f2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 10px;
}

<div
  className="c0"
>
  <h3
    className=""
  >
    Title
  </h3>
  <span
    className=""
  >
    Stars
  </span>
</div>
`;

exports[`With Snapshot Testing Should match snapshot without compound components children 1`] = `"This is a Panel"`;
